### ISSUE TITLE
Flatten all gradients in an MLP to a tensor of rank 1 (i.e. 1D array)

### ISSUE THREAD

**Issue by namp** [tensorflow/tensorflow#12724]
Suppose the following Keras model:

```
model = Sequential()
model.add(Dense(512, activation='sigmoid', input_shape=(784,)))
model.add(Dense(10, activation='softmax'))
```
Obviously we can calculate the gradients by:

`grads = K.gradients(loss, params)
` 

which just calls:

`tf.gradients(loss, variables, colocate_gradients_with_ops=True)
`

This returns a list of tensors containing:

1) a tensor with 512x784 elements (input to hidden connections)
2) a tensor with the biases of the 512 units in the hidden layer
3) a tensor with 10x512 elements (hidden to output connections)
4) a tensor with the biases of the 10 output units

I would like to ask if there's a simple way to "flatten" `grads` to a single tensor of rank 1 (i.e. 1D array) with `(512x784)+512+(10x512)+10 `elements, without looping over the layers and corresponding biases.

Thanks

**Comment by michaelisard** [comment]
This question is better asked on  [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a  bug or feature request. There is also a larger community that reads questions there. Thanks!

**Comment by namp** [comment]
Yes you are right - sorry about that. I was just hoping to reach more tensorflow experts here
